app/lib/katello/validators/root_repository_unique_attribute_validator.rb

```diff
@@ -2,7 +2,7 @@ module Katello
   module Validators
     class RootRepositoryUniqueAttributeValidator < ActiveModel::EachValidator
       def validate_each(record, attribute, value)
-        exists = RootRepository.where(attribute => value).where("id != ?", record.id || -1).exists?
+        exists = RootRepository.where(:product_id => record.product_id, attribute => value).where("id != ?", record.id || -1).exists?
```

Comment:
Did you know you can pass non-string where conditions to [`exists?`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-exists-3F) (also shown is refactoring of id to not use a string)?

i.e.

```rb
exists = RootRepository.where.not(id: record.id || -1).exists?(:product_id => record.product_id, attribute => value)
```